app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientAccountsFragment.java

```diff
@@ -182,22 +195,196 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
         inflater.inflate(R.menu.menu_account, menu);
         if (viewPager.getCurrentItem() == 0) {
             menu.findItem(R.id.menu_add_loan).setVisible(false);
+            menu.findItem(R.id.menu_filter_savings).setVisible(true);
+            menu.findItem(R.id.menu_filter_loan).setVisible(false);
+            menu.findItem(R.id.menu_filter_shares).setVisible(false);
+            menu.findItem(R.id.menu_search_saving).setVisible(true);
+            menu.findItem(R.id.menu_search_loan).setVisible(false);
+            menu.findItem(R.id.menu_search_share).setVisible(false);
+            initSearchSavings(menu);
         } else if (viewPager.getCurrentItem() == 1) {
             menu.findItem(R.id.menu_add_loan).setVisible(true);
+            menu.findItem(R.id.menu_filter_savings).setVisible(false);
+            menu.findItem(R.id.menu_filter_loan).setVisible(true);
+            menu.findItem(R.id.menu_filter_shares).setVisible(false);
+            menu.findItem(R.id.menu_search_saving).setVisible(false);
+            menu.findItem(R.id.menu_search_loan).setVisible(true);
+            menu.findItem(R.id.menu_search_share).setVisible(false);
+            initSearchLoans(menu);
         } else if (viewPager.getCurrentItem() == 2) {
             menu.findItem(R.id.menu_add_loan).setVisible(false);
+            menu.findItem(R.id.menu_filter_savings).setVisible(false);
+            menu.findItem(R.id.menu_filter_loan).setVisible(false);
+            menu.findItem(R.id.menu_filter_shares).setVisible(true);
+            menu.findItem(R.id.menu_search_saving).setVisible(false);
+            menu.findItem(R.id.menu_search_loan).setVisible(false);
+            menu.findItem(R.id.menu_search_share).setVisible(true);
+            initSearchShares(menu);
         }
         super.onCreateOptionsMenu(menu, inflater);
     }
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
+            case R.id.menu_filter_savings:
+                showSavingsFilterDialog();
+                break;
+            case R.id.menu_filter_loan:
+                showLoanFilterDialog();
+                break;
+            case R.id.menu_filter_shares:
+                showShareFilterDialog();
+                break;
             case R.id.menu_add_loan:
                 ((BaseActivity) getActivity()).replaceFragment(LoanApplicationFragment.
                         newInstance(LoanState.CREATE), true, R.id.container);
                 break;
         }
         return true;
     }
+
+    private void initSearchSavings(Menu menu) {
```

Comment:
@dilpreet96 you can make the single generic type of method, By just passing 2 -3 parameter and if statements and can use anywhere. Please do